podpac/core/cache/cache.py

```diff
@@ -1180,6 +1254,10 @@ def _get_key(self, obj):
     def _get_full_key(self, node, coordinates, key):
         return (self._get_key(node), self._get_key(coordinates), key)
 
+    def size(self):
+        process = psutil.Process(os.getpid())
+        return process.memory_info().rss # this is actually the total size of the process
```

Comment:
I think this is okay, but do you want to measure the actual ram cache size, instead? We can do that.

Reply:
That's a good question. For the use cases that come to mind, I think limiting the entire process would be desirable, but limiting via cache put requests doesn't necessarily accomplish this. Someone could easily exceed the limits without even using the cache.

How do we limit the cache size? I couldn't find a good universal way to determine the size of arbitrary data objects. Would we need to restrict the types of objects that get put into the cache? 

We could serialize the objects (with pickle) and determine the size of the resulting string, but this will generally take up more space then the actual object. Then again storing the objects serialized would allow for inter-process sharing, but we would take CPU/RAM hit serializing/de-serializing everything.

@mpu-creare  any thoughts?

Reply:
> I couldn't find a good universal way to determine the size of arbitrary data objects.

I think you can do good enough, see maybe https://stackoverflow.com/a/40880923/484488

> That's a good question. For the use cases that come to mind, I think limiting the entire process would be desirable, but limiting via cache put requests doesn't necessarily accomplish this. Someone could easily exceed the limits without even using the cache.

I don't exactly disagree, but this hardly accomplishes that. If we want to limit the entire process RAM usage, we need to (1) name the setting `RAM_MAX_BYTES` and (2) somehow also check and prevent *non-cache* objects from being added to RAM.

Reply:
Oh, I misread your comment, even as I quoted it! I don't agree that *"limiting the entire process would be desirable"*, but I do agree that *"limiting via cache put requests doesn't ~~necessarily~~ accomplish this"*.

Reply:
Thanks @jmilloy , let me check in with @mpu-creare about priorities. He's in a meeting. I''l hold for now on the merge.

Reply:
I like the current approach as implemented. We just need to carefully document it. 
I agree that it might be a little surprising to users, but I think it will be more maintainable in the long run and it's what I prefer for my applications as well. 

I don't like the other approach because in my experience it can be pretty fragile. Especially with all of the in-memory/out-of-memory operations that xarray does with different file formats. 

Also, by limiting the cache based on the entire process RAM, you have a hope of limiting to total RAM of the process (at least the cache won't make it worse than you've specified).

Reply:
I'll update the [settings docs](https://github.com/creare-com/podpac/blob/develop/doc/source/user/settings.md) before merging.

Reply:
It is the simplest implementation, and I think it's okay.

But imagine that you have "delete oldest" as your cache policy. You exceed the max ram settings outside of the cache. Then you try to put something in the cache, and the result is that your *entire* RAM cache is deleted. Now your executes very slowly. It would be very difficult to figure out why.

> the in-memory/out-of-memory operations that xarray does with different file formats.

Does this not effect either approach?

In any case, the default should be much larger, maybe the 100MB that you originally suggested, or even 1GB, if we're going to use the setting this way.